### PR TITLE
feat: Add haxe parser and queries.

### PR DESCRIPTION
**Pull request by vantreeseba**

### CODE REVIEW

File: lua/nvim-treesitter/parsers.lua [hunk 1]
```diff
@@ -659,6 +659,15 @@ list.haskell_persistent = {
   maintainers = { "@lykahb" },
 }
 
+list.haxe = {
+  install_info = {
+    url = "https://github.com/vantreeseba/tree-sitter-haxe",
+    files = { "src/parser.c" },
+  },
+  filetype = "haxe",
```
Comment (ObserverOfTime) [inline lua/nvim-treesitter/parsers.lua:667]: This is unnecessary.

```suggestion
```

File: queries/haxe/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,96 @@
+(identifier !type) @variable
+(comment) @comment
+
+; Preprocessor Statement
+; --------
+(preprocessor_statement) @tag
```
Comment (ObserverOfTime) [inline queries/haxe/highlights.scm:6]:
```suggestion
(preprocessor_statement) @preproc
```

File: queries/haxe/injections.scm [hunk 3]
```diff
@@ -0,0 +1,3 @@
+(interpolation) @haxe
+(comment) @jsdoc
```
Comment (ObserverOfTime) [inline queries/haxe/injections.scm:2]:
```suggestion

((comment) @jsdoc
 (#lua-match? @jsdoc "^/[*][*][^*].*[*]/$"))
  
```

File: queries/haxe/highlights.scm [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  "${" @punctuation.special
+  "}" @punctuation.special
+) @embedded
+
+
+; Tokens
+; ------
+
+(":") @punctuation.special
+(pair [":" "=>"] @punctuation.special)
```
Comment (ObserverOfTime) [inline queries/haxe/highlights.scm:80]: Assuming "@" is a token.

```suggestion
[ "@" ":" ] @punctuation.special
(pair "=>" @punctuation.special)
```

File: queries/haxe/highlights.scm [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+(bool) @constant
+(operator) @operator
+
+; Interpolation
+; -------------
+(interpolation "$" @punctuation.special)
+(interpolation
+  "${" @punctuation.special
+  "}" @punctuation.special
+) @embedded
```
Comment (ObserverOfTime) [inline queries/haxe/highlights.scm:73]: We don't have `@embedded`. This should work.

```suggestion
) @none
```

**Review by ObserverOfTime — CHANGES_REQUESTED** [review]

File: queries/haxe/highlights.scm [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+; TODO: Figure out how to determined when "nested member call" is last ident.
+; apparently this is a known issue https://github.com/tree-sitter/tree-sitter/issues/880
+(call_expression object: [
+  (identifier) @function 
+  (_ (identifier) @method .)
+  (_(_ (identifier) @method .))
+  (_(_(_ (identifier) @method .)))
+  (_(_(_(_ (identifier) @method .))))
+  (_(_(_(_(_ (identifier) @method .)))))
```
Comment (lucario387) [inline queries/haxe/highlights.scm:53]: I know that this is a way to workaround with issue from upstream, but I'd prefer these to be commented out in nvim-treesitter queries. You can add it back from local config anyways.

File: queries/haxe/tags.scm
Comment (lucario387) [inline queries/haxe/tags.scm:1]: Please remove.

File: tests/query/highlights/haxe/class.hx [hunk 7]
```diff
@@ -0,0 +1,9 @@
+class Main {}
+// <- keyword
+//    ^ variable
```
Comment (lucario387) [inline tests/query/highlights/haxe/class.hx:3]: Shouldn't this be `@type`? also for interfaces

File: queries/haxe/locals.scm [hunk 8]
```diff
@@ -0,0 +1,12 @@
+; Scopes
+[
+ (block)
+ (function_declaration)
+] @scope @local.scope
```
Comment (clason) [inline queries/haxe/locals.scm:5]: We don't use the upstream format (yet).

**Comment by lucario387** [comment]
Please don't bring the entire test suite from `tree-sitter-haxe` to here. Instead, bring only tests that reflect specific changes to accomodate with `nvim-treesitter` differences.

**Review by clason — COMMENTED** [review]
Since you seem to be copying the upstream queries verbatim: Are you aware that Neovim uses a different query order precedence ("last match wins") than upstream ("first match wins")?

**Comment by vantreeseba** [comment]
Thanks for all the feedback / reviews, will try to get to those soon.

**Comment by clason** [comment]
We've just switched the format of injection queries to that of upstream (see CONTRIBUTING.md). You'll have to rebase on (not merge in!) `master` and adapt `injections.scm`, sorry.

**Comment by clason** [comment]
Closing as stale; if people still want this, it's better to start from scratch based on the current state of the parser and our contributing guidelines.

**Pull request closed without merging**